fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java

```diff
@@ -471,62 +472,116 @@ public TDescribeTableResult describeTable(TDescribeTableParams params) throws TE
-                            desc.setColumnKey("");
-                        }
-                        final TColumnDef colDef = new TColumnDef(desc);
-                        final String comment = column.getComment();
-                        if (comment != null) {
-                            colDef.setComment(comment);
-                        }
-                        columns.add(colDef);
-                        // if user set limit, then only return limit size result
-                        if (limit > 0 && columns.size() >= limit) {
-                            break;
-                        }
-                    }
-                }
+                setColumnDesc(columns, table, limit, false, params.db, params.getTable_name());
             } finally {
                 db.readUnlock();
             }
         }
         return result;
     }
 
+    // get describeTable without db name and table name parameter, so we need iterate over
+    // dbs and tables, when reach limit, we break;
+    private void describeWithoutDbAndTable(UserIdentity currentUser, List<TColumnDef> columns, long limit) {
+        Catalog catalog = Catalog.getCurrentCatalog();
+        List<String> dbNames = catalog.getDbNames();
+        boolean reachLimit;
+        for (String fullName : dbNames) {
+            if (!Catalog.getCurrentCatalog().getAuth().checkDbPriv(currentUser, fullName, PrivPredicate.SHOW)) {
+                continue;
+            }
+            String dbName = fullName.indexOf(":") >= 0 ? fullName.substring(fullName.indexOf(":") + 1) : fullName;
+            Database db = Catalog.getCurrentCatalog().getDb(fullName);
+            if (db != null) {
+                for (String tableName : db.getTableNamesWithLock()) {
+                    LOG.debug("get table: {}, wait to check", tableName);
+                    if (!Catalog.getCurrentCatalog().getAuth().checkTblPriv(currentUser, dbName,
+                            tableName, PrivPredicate.SHOW)) {
+                        continue;
+                    }
+                    db.readLock();
+                    try {
+                        Table table = db.getTable(tableName);
+                        reachLimit = setColumnDesc(columns, table, limit, true, dbName, tableName);
+                    } finally {
+                        db.readUnlock();
+                    }
+                    if (reachLimit) {
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean setColumnDesc(List<TColumnDef> columns, Table table, long limit,
+                               boolean needSetDbAndTable, String db, String tbl) {
+        if (table != null) {
+            String tableKeysType = "";
+            if (TableType.OLAP.equals(table.getType())) {
+                OlapTable olapTable = (OlapTable) table;
+                tableKeysType = olapTable.getKeysType().name().substring(0, 3).toUpperCase();
```

Comment:
What does it means by `substring(0, 3)`?

Reply:
This is the previous code logic, which has not been changed. My understanding is to get values ​​like "PRI, UNI, DUP, AGG".